docs/concepts/configuration/pod-priority-preemption.md

```diff
@@ -0,0 +1,197 @@
+when they should use this PriorityClass.
+
+
+**Note 1:** If you upgrade your existing cluster and enable this feature, the priority
+of your existing pods will be considered to be zero.
+
+**Note 2:** Addition of a PriorityClass with `globalDefault` set to true does not
+change priority of existing pods. The value of such PriorityClass will be used only
+for pods created after the PriorityClass is added.
+
+**Note 3:** If you delete a PriorityClass, existing pods that use the name of the
+deleted priority class will remain unchanged, but you will not be able to create more pods
+that use the name of the deleted priority class.
+
+#### Example PriorityClass
+```yaml
+apiVersion: v1
+kind: PriorityClass
+metadata:
+  name: high-priority
+value: 1000000
+globalDefault: false
+description: "This priority class should be used for XYZ service pods only."
+```
+
+## Pod Priority
+Once you have one or more PriorityClasses, you can create pods which specify one
+of those PriorityClass names in their spec. Priority admission controller uses
+`priorityClassName` field and populates the integer value of priority. If the priority
+class is not found, the pod will be rejected.
+
+The following YAML is an example of a pod configuration that uses the PriorityClass
+created above. Priority admission controller checks the spec and resolves the
+priority of the pod to 1000000.
+
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx
+  labels:
+    env: test
+spec:
+  containers:
+  - name: nginx
+    image: nginx
+    imagePullPolicy: IfNotPresent
+  priorityClassName: high-priority
+```
+
+## Preemption
+When pods are created, they go to a queue and wait to be scheduled. Scheduler picks a pod
+from the queue and tries to schedule it on a node. If no node is found that satisfies
+all the specified requirements (predicates) of the pod, preemption logic is triggered 
+for the pending pod. Let's call the pending pod P.
+Preemption logic tries to find a node where removal of one or more pods with lower priority 
+than P would enable P to schedule on that node. If such a node is found, one or more lower priority pods will
+be deleted from the node. Once the pods are gone, P may be scheduled on the node. 
+
+### Limitations of Preemption (alpha version)
+
+#### Starvation of Preempting Pod
+When pods are preempted, the victims get their
+[graceful termination period](https://kubernetes.io/docs/concepts/workloads/pods/pod/#termination-of-pods).
+They have that much time to finish their work and exit. If they don't, they will be
+killed. This graceful termination period creates a time gap between the point that
+scheduler preempts pods until the pending pod (P) can be scheduled on the node (N).
+In the meantime, scheduler keeps scheduling other pending pods. When one or more victims
+exit or get terminated, scheduler may place other pending pods on the node if the pods are
+ahead of P in the scheduling queue. In such a case, it is likely that
```

Comment:
IIUC "ahead of P in the scheduling queue" isn't really relevant. It's just whichever pod is next in the scheduling queue when the victim terminates. It might be ahead of P or behind P. I would change this sentence to "As victims exit or get terminated, scheduler continues trying to schedule all of the pods in the pending queue, and one or more of them may be considered and scheduled to N before the scheduler considers scheduling P on N."

Reply:
Other pods must be ahead of P when victim exits or get terminated, otherwise P will be scheduled first. Anyway, I used almost the same text as you suggested.

Reply:
I think we're just using definitions of "ahead of P." To me, "ahead of" means something is to the left of something else in a queue. But this is a circular queue and when you say "ahead of" you really mean "processed before" -- it may actually be to the right. Anyway, the new wording LGTM.